server/routes/prisonerChangeIncentiveLevelDetails.test.ts

```diff
@@ -0,0 +1,324 @@
+          expect(res.text).toContain('Enter a reason for recording')
+        })
+    })
+
+    it('should return an error if form is missing an incentive level', () => {
+      return request(app)
+        .post(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+        .set('authorization', `bearer ${tokenWithNecessaryRole}`)
+        .send({
+          newIepLevel: '',
+          reason: 'reason',
+        })
+        .expect(200)
+        .expect(res => {
+          expect(res.text).toContain('Select an incentive level, even if it is the same as before')
+          expect(res.text).not.toContain('Enter a reason for recording')
+          expect(res.text).not.toContain('Comments must be 240 characters or less')
+        })
+    })
+
+    it('should return an error if form is missing a reason', () => {
+      return request(app)
+        .post(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+        .set('authorization', `bearer ${tokenWithNecessaryRole}`)
+        .send({
+          newIepLevel: 'STD',
+          reason: '',
+        })
+        .expect(200)
+        .expect(res => {
+          expect(res.text).toContain('Enter a reason for recording')
+          expect(res.text).not.toContain('Select an incentive level')
+        })
+    })
+
+    it('should return an error if comment is over 240 characters', () => {
+      return request(app)
+        .post(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+        .set('authorization', `bearer ${tokenWithNecessaryRole}`)
+        .send({
+          newIepLevel: 'STD',
+          reason:
+            'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. ' +
+            'Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. ' +
+            'Donec quam felis, ultricies nec, pellentesque eu,',
+        })
+        .expect(200)
+        .expect(res => {
+          expect(res.text).toContain('Comments must be 240 characters or less')
+          expect(res.text).not.toContain('Select an incentive level, even if it is the same as before')
+          expect(res.text).not.toContain('Enter a reason for recording')
+        })
+    })
+
+    it('should retain the inputted form values', () => {
+      return request(app)
+        .post(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+        .set('authorization', `bearer ${tokenWithNecessaryRole}`)
+        .send({
+          newIepLevel: '',
+          reason: 'Should retain this text',
+        })
+        .expect(200)
+        .expect(res => {
+          expect(res.text).toContain('Select an incentive level, even if it is the same as before')
+          expect(res.text).toContain('Should retain this text')
+        })
+    })
+
+    it('should return generic error page if api returns an error', () => {
```

Comment:
good idea to test this, actually. i would recommend something other than a 404 though – there are too many other potential causes of a 404. maybe a 400 just to be different. i would have it only on the `updateIncentiveLevelForPrisoner` so that we're pretending the actual form being posted was the cause of the error.